dist: support smooth upgrade from enterprise to source available

yaronkaikov opened the pull request:
When upgrading for example from `2024.1` to `2025.1` the package name is not identically causing the upgrade command to fail:
```
Command: 'sudo DEBIAN_FRONTEND=noninteractive apt-get dist-upgrade scylla -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold"'
Exit code: 100
Stdout:
Selecting previously unselected package scylla.
Preparing to unpack .../6-scylla_2025.1.0~dev-0.20250118.1ef2d9d07692-1_amd64.deb ...
Unpacking scylla (2025.1.0~dev-0.20250118.1ef2d9d07692-1) ...
Errors were encountered while processing:
/tmp/apt-dpkg-install-JbOMav/0-scylla-conf_2025.1.0~dev-0.20250118.1ef2d9d07692-1_amd64.deb
/tmp/apt-dpkg-install-JbOMav/1-scylla-python3_2025.1.0~dev-0.20250118.1ef2d9d07692-1_amd64.deb
/tmp/apt-dpkg-install-JbOMav/2-scylla-server_2025.1.0~dev-0.20250118.1ef2d9d07692-1_amd64.deb
/tmp/apt-dpkg-install-JbOMav/3-scylla-kernel-conf_2025.1.0~dev-0.20250118.1ef2d9d07692-1_amd64.deb
/tmp/apt-dpkg-install-JbOMav/4-scylla-node-exporter_2025.1.0~dev-0.20250118.1ef2d9d07692-1_amd64.deb
/tmp/apt-dpkg-install-JbOMav/5-scylla-cqlsh_2025.1.0~dev-0.20250118.1ef2d9d07692-1_amd64.deb
Stderr:
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Adding `Obsoletes` (for rpm) and `Replaces` (for deb)

Fixes: https://github.com/scylladb/scylladb/issues/22420

**Source available changes, since we branched to 2025.1 it requires a backport for upgrades to work**

### Testing
I have performed the upgrade process using a docker image, attaching the logs for both testing
- [x] rpm upgrade - [rockylinux9_upgrade.log](https://github.com/user-attachments/files/18708637/rockylinux9_upgrade.log)
- [x] deb upgrade - [ubuntu22.04_upgrade.log](https://github.com/user-attachments/files/18708650/ubuntu22.04_upgrade.log)

avikivity commented on dist/debian/control.template:
```diff
@@ -13,12 +13,14 @@ Description: Scylla database main configuration file
 Replaces: %{product}-tools (<<5.5)
+Replaces: %{product}-enterprise-tools (<<5.5)
 Breaks: %{product}-tools (<<5.5)
 Description: Scylla database server binaries 
```
What about the other packages? like scylla-enterprise, scylla-enterprise-cqlsh?

avikivity replied:
And why 5.5?

yaronkaikov replied:
> What about the other packages? like scylla-enterprise, scylla-enterprise-cqlsh?

I would expect it to be there before if needed, right?

avikivity commented on dist/redhat/scylla.spec:
```diff
@@ -13,7 +13,13 @@ Requires:       %{product}-python3 = %{version}-%{release}
+Obsoletes:      scylla-enterprise-python3
+Obsoletes:      scylla-enterpise-kernel-conf
+Obsoletes:      scylla-enterpise-node-exporter
+Obsoletes:      scylla-enterpise-cqlsh
```
Seems wrong. What if I'm not installing the scylla package, just scylla-server?

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:x: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14655/)
#### Build Details:
- Duration: 35 min
- Builder: spider4.cloudius-systems.com

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14662/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1133/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/6066/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2944/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/14077/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/568/)
:white_check_mark: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/72/)
:white_check_mark: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/73/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14662/)
#### Build Details:
- Duration: 10 hr
- Builder: i-04f8b03e13830eb04 (m5ad.8xlarge)

fruch commented:
@yaronkaikov 

What about 
scylla-python3 and scylla-cqlsh ?

yaronkaikov commented:
> @yaronkaikov
> 
> What about scylla-python3 and scylla-cqlsh ?

Added, although i am not sure how the rolling upgrade did pass

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14681/)
:white_check_mark: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/73/)
:x: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2954/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/6077/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/14087/)
:white_check_mark: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/74/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1134/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/569/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14681/)
#### Failed Tests (4/37272):
- [test_trigger_snapshot_transfer](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14681/testReport/junit/topology_operation_during_upgrade_test/TopologyOperationWithMixedCLusterTest/Tests___dtest_with_tablets___test_trigger_snapshot_transfer) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_trigger_snapshot_transfer)
- [test_rolling_upgrade](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14681/testReport/junit/enterprise_upgrade_test/TestEnterpriseRollingUpgrade/Tests___dtest_with_topology_changes___test_rolling_upgrade) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_rolling_upgrade)
- [test_rolling_upgrade](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14681/testReport/junit/enterprise_upgrade_test/TestEnterpriseRollingUpgrade/Tests___dtest_with_tablets___test_rolling_upgrade) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_rolling_upgrade)
- [test_rolling_upgrade](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14681/testReport/junit/enterprise_upgrade_test/TestEnterpriseRollingUpgrade/Tests___dtest___test_rolling_upgrade) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_rolling_upgrade)
#### Build Details:
- Duration: 4 hr 46 min
- Builder: spider3.cloudius-systems.com

avikivity commented:
Please add to the patch changelog a description of the tests done to make sure this works.

fruch commented:
> > @yaronkaikov
> > What about scylla-python3 and scylla-cqlsh ?
> 
> Added, although i am not sure how the rolling upgrade did pass

it did not

```
/tmp/apt-dpkg-install-grlcLT/0-scylla-conf_2025.1.0~dev-0.20250122.025d64150946-1_amd64.deb
/tmp/apt-dpkg-install-grlcLT/1-scylla-python3_2025.1.0~dev-0.20250122.025d64150946-1_amd64.deb
/tmp/apt-dpkg-install-grlcLT/2-scylla-server_2025.1.0~dev-0.20250122.025d64150946-1_amd64.deb
/tmp/apt-dpkg-install-grlcLT/3-scylla-kernel-conf_2025.1.0~dev-0.20250122.025d64150946-1_amd64.deb
/tmp/apt-dpkg-install-grlcLT/4-scylla-node-exporter_2025.1.0~dev-0.20250122.025d64150946-1_amd64.deb
/tmp/apt-dpkg-install-grlcLT/5-scylla-cqlsh_2025.1.0~dev-0.20250122.025d64150946-1_amd64.deb
Stderr:
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

see:
https://argus.scylladb.com/tests/scylla-cluster-tests/8f4fce27-50d8-4e96-b0c2-1de6e6485adc

tchaikov commented on dist/debian/control.template:
```diff
@@ -12,13 +12,13 @@ Architecture: any
 Architecture: any
 Depends: ${misc:Depends}, %{product}-conf (= ${binary:Version}), %{product}-python3 (= ${binary:Version})
-Replaces: %{product}-tools (<<5.5)
+Replaces: %{product}-tools (<<5.5), %{product}-enterprise-tools (<<2024.2), %{product}-enterprise-python3 (<<2024.2), %{product}-enterprise-cqlsh (<<2024.2)
```
why does the new scylla-server package "replaces"  scylla-enterprise-tool? BTW, i don't think it makes sense to use "${product}-enterprise" here, "scylla-enterprise" is a product name. and we are replacing the packages with these *fixed* names, not a set of packages whose name can be generated from this product -- "scylla".

tchaikov commented:
> @yaronkaikov
> 
> What about scylla-python3 and scylla-cqlsh ?

i think we should update the submodule's debian and rpm recipe respectively.

syuu1228 commented on dist/redhat/scylla.spec:
```diff
@@ -13,7 +13,13 @@ Requires:       %{product}-python3 = %{version}-%{release}
+Obsoletes:      scylla-enterpise-kernel-conf <= %{version}-%{release}
+Obsoletes:      scylla-enterpise-node-exporter <= %{version}-%{release}
+Obsoletes:      scylla-enterpise-cqlsh <= %{version}-%{release}
+
```
I think this should be rewritten like this.
`Obsoletes` need to be define on each subpackages, and version should be specified like `< 2025.1.0`, instead of using version and release parameters.
Because on Fedora doc: https://docs.fedoraproject.org/en-US/packaging-guidelines/#renaming-or-replacing-existing-packages
It suggests:
_You usually do not use macros for this as you’re simply trying to advance beyond the last known release under the old name._

And also we need to `Provides` to provide enterprise virtual package.

Additionally we need same fix on tools/cqlsh, tools/python3.
```
diff --git a/dist/redhat/scylla.spec b/dist/redhat/scylla.spec
index dc7ebf590a..bb13c8991b 100644
--- a/dist/redhat/scylla.spec
+++ b/dist/redhat/scylla.spec
@@ -13,12 +13,8 @@ Requires:       %{product}-python3 = %{version}-%{release}
 Requires:       %{product}-kernel-conf = %{version}-%{release}
 Requires:       %{product}-node-exporter = %{version}-%{release}
 Requires:       %{product}-cqlsh = %{version}-%{release}
-Obsoletes:      scylla-enterpise-server <= %{version}-%{release}
-Obsoletes:      scylla-enterpise-conf <= %{version}-%{release}
-Obsoletes:      scylla-enterprise-python3 <= %{version}-%{release}
-Obsoletes:      scylla-enterpise-kernel-conf <= %{version}-%{release}
-Obsoletes:      scylla-enterpise-node-exporter <= %{version}-%{release}
-Obsoletes:      scylla-enterpise-cqlsh <= %{version}-%{release}
+Provides:       scylla-enterprise = %{version}-%{release}
+Obsoletes:      scylla-enterprise < 2025.1.0
 
 
 %global _debugsource_template %{nil}
@@ -79,6 +75,8 @@ Requires:       %{product}-python3 = %{version}-%{release}
 AutoReqProv:    no
 Provides:       %{product}-tools:%{_bindir}/nodetool
 Provides:       %{product}-tools:%{_sysconfigdir}/bash_completion.d/nodetool-completion
+Provides:       scylla-enterprise-server = %{version}-%{release}
+Obsoletes:      scylla-enterprise-server < 2025.1.0
 
 %description server
 This package contains ScyllaDB server.
@@ -162,6 +160,8 @@ ln -sfT /etc/scylla /var/lib/scylla/conf
 Group:          Applications/Databases
 Summary:        Scylla configuration package
 Obsoletes:      scylla-server < 1.1
+Provides:       scylla-enterprise-conf = %{version}-%{release}
+Obsoletes:      scylla-enterprise-conf < 2025.1.0
 
 %description conf
 This package contains the main scylla configuration file.
@@ -182,6 +182,8 @@ Summary:        Scylla configuration package for the Linux kernel
 Requires:       kmod
 # tuned overwrites our sysctl settings
 Obsoletes:      tuned >= 2.11.0
+Provides:       scylla-enterprise-kernel-conf = %{version}-%{release}
+Obsoletes:      scylla-enterprise-kernel-conf < 2025.1.0
 
 %description kernel-conf
 This package contains Linux kernel configuration changes for the Scylla database.  Install this package
@@ -218,6 +220,8 @@ Group:          Applications/Databases
 Summary:        Prometheus exporter for machine metrics
 License:        ASL 2.0
 URL:            https://github.com/prometheus/node_exporter
+Provides:       scylla-enterprise-node-exporter = %{version}-%{release}
+Obsoletes:      scylla-enterprise-node-exporter < 2025.1.0
 
 %description node-exporter
 Prometheus exporter for machine metrics, written in Go with pluggable metric collectors.
```

yaronkaikov replied:
implemented as suggested

syuu1228 commented on dist/debian/control.template:
```diff
@@ -12,13 +12,13 @@ Architecture: any
  Scylla is a highly scalable, eventually consistent, distributed,
  partitioned row DB.
-Replaces: %{product}-server (<< 1.1)
+Replaces: %{product}-server (<< 1.1), %{product}-enterprise-server (<<2024.2)
```
Some of comments might be already suggested by @tchaikov but I'm comment it anyway,

I think this should be rewritten like this.
Version number on Replaces should be `(<< 2025.1.0~)`, because <<2024.2 will not match with scylla-enterprise-2024.2.x.
And also need to add Breaks as tchaikov suggested.

We also need dummy enterprise package to make them able to upgrade by `apt-get dist-upgrade`, which is suggested on Debian wiki:
https://wiki.debian.org/RenamingPackages
It seems like `apt-get` does not able to detect packages are upgradable, without this.
Dummy packages are able to remove `apt-get autoremove` or `apt-get autopurge` after upgrade.

Additionally we need same fix on tools/cqlsh, tools/python3.
```
diff --git a/dist/debian/control.template b/dist/debian/control.template
index 88407eb8e0..ae279bfb54 100644
--- a/dist/debian/control.template
+++ b/dist/debian/control.template
@@ -12,14 +12,15 @@ Architecture: any
 Description: Scylla database main configuration file
  Scylla is a highly scalable, eventually consistent, distributed,
  partitioned row DB.
-Replaces: %{product}-server (<< 1.1), %{product}-enterprise-server (<<2024.2)
+Replaces: %{product}-server (<< 1.1), scylla-enterprise-conf (<< 2025.1.0~)
 Conflicts: %{product}-server (<< 1.1)
+Breaks: scylla-enterprise-conf (<< 2025.1.0~)
 
 Package: %{product}-server
 Architecture: any
 Depends: ${misc:Depends}, %{product}-conf (= ${binary:Version}), %{product}-python3 (= ${binary:Version})
-Replaces: %{product}-tools (<<5.5), %{product}-enterprise-tools (<<2024.2), %{product}-enterprise-python3 (<<2024.2), %{product}-enterprise-cqlsh (<<2024.2)
-Breaks: %{product}-tools (<<5.5)
+Replaces: %{product}-tools (<<5.5), scylla-enterprise-server (<< 2025.1.0~)
+Breaks: %{product}-tools (<<5.5), scylla-enterprise-server (<< 2025.1.0~)
 Description: Scylla database server binaries 
  Scylla is a highly scalable, eventually consistent, distributed,
  partitioned row DB.
@@ -29,6 +30,8 @@ Section: debug
 Priority: extra
 Architecture: any
 Depends: %{product}-server (= ${binary:Version}), ${misc:Depends}
+Replaces: scylla-enterprise-server-dbg (<< 2025.1.0~)
+Breaks: scylla-enterprise-server-dbg (<< 2025.1.0~)
 Description: debugging symbols for %{product}-server
  Scylla is a highly scalable, eventually consistent, distributed,
  partitioned row DB.
@@ -37,13 +40,17 @@ Description: debugging symbols for %{product}-server
 Package: %{product}-kernel-conf
 Architecture: any
 Depends: procps
+Replaces: scylla-enterprise-kernel-conf (<< 2025.1.0~)
+Breaks: scylla-enterprise-kernel-conf (<< 2025.1.0~)
 Description: Scylla kernel tuning configuration
  Scylla is a highly scalable, eventually consistent, distributed,
  partitioned row DB.
 
 Package: %{product}-node-exporter
 Architecture: any
+Replaces: scylla-enterprise-node-exporter (<< 2025.1.0~)
 Conflicts: prometheus-node-exporter
+Breaks: scylla-enterprise-node-exporter (<< 2025.1.0~)
 Description: Prometheus exporter for machine metrics
  Prometheus exporter for machine metrics, written in Go with pluggable metric collectors.
 
@@ -54,6 +61,57 @@ Depends: %{product}-server (= ${binary:Version})
  , %{product}-kernel-conf (= ${binary:Version})
  , %{product}-node-exporter (= ${binary:Version})
  , %{product}-cqlsh (= ${binary:Version})
+Replaces: scylla-enterprise (<< 2025.1.0~)
+Breaks: scylla-enterprise (<< 2025.1.0~)
 Description: Scylla database metapackage
  Scylla is a highly scalable, eventually consistent, distributed,
  partitioned row DB.
+
+
+Package: scylla-enterprise-conf
+Depends: %{product}-conf (= ${binary:Version})
+Architecture: all
+Priority: optional
+Section: oldlibs
+Description: transitional package
+ This is a transitional package. It can safely be removed.
+
+Package: scylla-enterprise-server
+Depends: %{product}-server (= ${binary:Version})
+Architecture: all
+Priority: optional
+Section: oldlibs
+Description: transitional package
+ This is a transitional package. It can safely be removed.
+
+Package: scylla-enterprise-server-dbg
+Depends: %{product}-server-dbg (= ${binary:Version})
+Architecture: all
+Priority: optional
+Section: oldlibs
+Description: transitional package
+ This is a transitional package. It can safely be removed.
+
+Package: scylla-enterprise-kernel-conf
+Depends: %{product}-kernel-conf (= ${binary:Version})
+Architecture: all
+Priority: optional
+Section: oldlibs
+Description: transitional package
+ This is a transitional package. It can safely be removed.
+
+Package: scylla-enterprise-node-exporter
+Depends: %{product}-node-exporter (= ${binary:Version})
+Architecture: all
+Priority: optional
+Section: oldlibs
+Description: transitional package
+ This is a transitional package. It can safely be removed.
+
+Package: scylla-enterprise
+Depends: %{product} (= ${binary:Version})
+Architecture: all
+Priority: optional
+Section: oldlibs
+Description: transitional package
+ This is a transitional package. It can safely be removed.
```

syuu1228 commented:
Opened PRs for scylla-cqlsh and scylla-python3:
- https://github.com/scylladb/scylla-python3/pull/43
- https://github.com/scylladb/scylla-cqlsh/pull/121

yaronkaikov commented:
v2:
- applied changes based on @tchaikov and @syuu1228 suggestion

yaronkaikov commented:
> Opened PRs for scylla-cqlsh and scylla-python3:
> 
> * [dist: support smooth upgrade from enterprise to source availalbe scylla-python3#43](https://github.com/scylladb/scylla-python3/pull/43)
> * [dist: support smooth upgrade from enterprise to source availalbe scylla-cqlsh#121](https://github.com/scylladb/scylla-cqlsh/pull/121)

We need to wait for both to be promoted before we can verify

fruch commented:
@yaron can you take those submodule update into this PR, and then we could take it as a whole ?

yaronkaikov commented:
v3:
- added both python3 and cqlsh changes (i will drop it later) for debug

avikivity commented:
It's reasonable to commit the submodules at the same time (if needed).

yaronkaikov commented:
> It's reasonable to commit the submodules at the same time (if needed).

Ok, i have only added them for debug, if everything will work, we can submit them together

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14741/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1136/)
:white_check_mark: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/85/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/3006/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/6127/)
:x: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/84/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/14138/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/580/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14741/)
#### Build Details:
- Duration: 10 hr
- Builder: i-0e8f0b7266435f6ce (m5ad.8xlarge)

fruch commented:
@yaronkaikov 

something is broken with the docker based artifact tests, sounds like installation didn't work, but logs are not uploaded to jenkins

syuu1228 commented on dist/redhat/scylla.spec:
```diff
@@ -13,7 +13,9 @@ Requires:       %{product}-python3 = %{version}-%{release}
 Requires:       %{product}-cqlsh = %{version}-%{release}
-Obsoletes:      scylla-server < 1.1
+Provides:	    scylla-enterprise = %{version}-%{release}
+Obsoletes:      scylla-enterpise < 2025.1.0
```
> scylla-enterpise
I think this is typo.

Should be:
```
-Obsoletes:      scylla-enterpise < 2025.1.0
+Obsoletes:      scylla-enterprise < 2025.1.0
```

yaronkaikov replied:
fixed

syuu1228 commented on dist/debian/control.template:
```diff
@@ -37,23 +40,49 @@ Description: debugging symbols for %{product}-server
-Depends: %{product}-server (= ${binary:Version})
- , %{product}-kernel-conf (= ${binary:Version})
- , %{product}-node-exporter (= ${binary:Version})
- , %{product}-cqlsh (= ${binary:Version})
```
We have to keep these lines, otherwise package installation break up.

syuu1228 commented on dist/debian/control.template:
```diff
@@ -37,23 +40,49 @@ Description: debugging symbols for %{product}-server
+Section: oldlibs
+Description: transitional package
+ This is a transitional package. It can safely be removed.
+
```
These transition packages are missing, this breaks upgrade.
```
Package: scylla-enterprise-server-dbg
Depends: %{product}-server-dbg (= ${binary:Version})
Architecture: all
Priority: optional
Section: oldlibs
Description: transitional package
 This is a transitional package. It can safely be removed.

Package: scylla-enterprise-kernel-conf
Depends: %{product}-kernel-conf (= ${binary:Version})
Architecture: all
Priority: optional
Section: oldlibs
Description: transitional package
 This is a transitional package. It can safely be removed.

Package: scylla-enterprise-node-exporter
Depends: %{product}-node-exporter (= ${binary:Version})
Architecture: all
Priority: optional
Section: oldlibs
Description: transitional package
 This is a transitional package. It can safely be removed.
```

yaronkaikov replied:
added

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14772/)
:x: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1138/)
:x: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/90/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/3033/)
:white_check_mark: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/91/)
:x: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/586/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/6155/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/14170/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14772/)
#### Build Details:
- Duration: 9 hr 37 min
- Builder: i-098f379d552df4652 (m5ad.12xlarge)

tchaikov commented on dist/debian/control.template:
```diff
@@ -54,6 +61,56 @@ Depends: %{product}-server (= ${binary:Version})
+ This is a transitional package. It can safely be removed.
+
+Package: scylla-enterprise-server-dbg
+Depends: %{product}-server-dbg (= ${binary:Version})
```
i don't think we should define transitional packages for -dbg packages. because 

- debug packages are development/debugging tools that aren't part of the regular runtime dependencies of other packages
- they're typically explicitly installed when needed rather than being pulled in as dependencies

yaronkaikov replied:
right, removed

yaronkaikov commented on dist/debian/control.template:
```diff
@@ -57,10 +57,6 @@ Description: Prometheus exporter for machine metrics
 Package: %{product}
```
done

denesb commented:
@yaronkaikov what is this test doing? Upgrade from which version to which one?

yaronkaikov commented:
> @yaronkaikov what is this test doing? Upgrade from which version to which one?

It is doing upgrade to Scylla , in this case from 2024.2 (the latest) to 2025.1.0-dev

denesb commented:
> > @yaronkaikov what is this test doing? Upgrade from which version to which one?
> 
> It is doing upgrade to Scylla , in this case from 2024.2 (the latest) to 2025.1.0-dev

Master is 2025.2~dev. That said 2025.1 already has file-stream. Maybe due to unsuccessful promotions, the last successful 2025.1~dev build is before the merge of file-stream.

yaronkaikov commented:
> > > @yaronkaikov what is this test doing? Upgrade from which version to which one?
> 
> > 
> 
> > It is doing upgrade to Scylla , in this case from 2024.2 (the latest) to 2025.1.0-dev
> 
> 
> 
> Master is 2025.2-dev. That said 2025.1 already has file-stream. Maybe due to unsuccessful promotions, the last successful 2025.1-dev build is before the merge of file-stream.

Oh , I think you are right . We need next to promote and then rebase

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14789/)
:x: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1140/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/6175/)
:x: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/95/)
:x: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/94/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/3053/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/14186/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/591/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14789/)
#### Failed Tests (3/37238):
- [test_delete_elements_from_a_set](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14789/testReport/junit/alternator_tests/TesterAlternator/Tests___dtest_with_topology_changes___test_delete_elements_from_a_set) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_delete_elements_from_a_set)
- [test_view_build_status_cleanup_on_remove_node.debug.3](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14789/testReport/junit/topology_custom/test_view_build_status/Tests___Unit_Tests___test_view_build_status_cleanup_on_remove_node_debug_3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_view_build_status_cleanup_on_remove_node.debug.3)
- [cqlpy.cassandra_tests/validation/entities/frozen_collections_test.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14789/testReport/junit/%28root%29/non-boost%20tests/Tests___Unit_Tests___cqlpy_cassandra_tests_validation_entities_frozen_collections_test_debug_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+cqlpy.cassandra_tests/validation/entities/frozen_collections_test.debug.1)
#### Build Details:
- Duration: 10 hr
- Builder: i-007222f3bd4df20fc (m5ad.8xlarge)

avikivity commented:
This will never reach completion if you test it using CI.

yaronkaikov commented:
> This will never reach completion if you test it using CI.

trying to verify it locally and manually, converting to DRAFT for now, so CI will not run

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14815/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1141/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/3069/)
:white_check_mark: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/98/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/14201/)
:white_check_mark: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/99/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/6190/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/595/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14815/)
#### Failed Tests (1/37244):
- [test_change_two.release.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14815/testReport/junit/topology/test_change_ip/Tests___Unit_Tests___test_change_two_release_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_change_two.release.1)
#### Build Details:
- Duration: 10 hr
- Builder: i-04edb8507a6fd716c (m5ad.8xlarge)

syuu1228 commented:
I was forgot to mention about one more thing, if we keep building scylla-enterprise and it will keep pulling from OSS repo, ALL of this changes requires `%if %{product} == "scylla"` - `%endif`. Otherwise it probably will break enterprise packages.

mykaul commented:
> I was forgot to mention about one more thing, if we keep building scylla-enterprise and it will keep pulling from OSS repo, ALL of this changes requires `%if %{product} == "scylla"` - `%endif`. Otherwise it probably will break enterprise packages.

So let's not keep building enterprise.

yaronkaikov commented:
> > I was forgot to mention about one more thing, if we keep building scylla-enterprise and it will keep pulling from OSS repo, ALL of this changes requires `%if %{product} == "scylla"` - `%endif`. Otherwise it probably will break enterprise packages.
> 
> So let's not keep building enterprise.

we are not, only gating

mykaul commented:
> > > I was forgot to mention about one more thing, if we keep building scylla-enterprise and it will keep pulling from OSS repo, ALL of this changes requires `%if %{product} == "scylla"` - `%endif`. Otherwise it probably will break enterprise packages.
> > 
> > 
> > So let's not keep building enterprise.
> 
> we are not, only gating

Gating on what?

yaronkaikov commented:
> > > > I was forgot to mention about one more thing, if we keep building scylla-enterprise and it will keep pulling from OSS repo, ALL of this changes requires `%if %{product} == "scylla"` - `%endif`. Otherwise it probably will break enterprise packages.
> > > 
> > > 
> > > So let's not keep building enterprise.
> > 
> > 
> > we are not, only gating
> 
> Gating on what?

enterprise

yaronkaikov commented:
@fruch https://jenkins.scylladb.com/job/scylla-staging/job/yaronkaikov/job/rolling-upgrade-ubuntu22.04-test/7/ passed successfully for 6.2, 2024.1 and 2024.2

Moving now to test the rpm part

syuu1228 commented:
dist/redhat/scylla.spec need to be fix like this:
```
diff --git a/dist/redhat/scylla.spec b/dist/redhat/scylla.spec
index 324818cac0..80c2322fe3 100644
--- a/dist/redhat/scylla.spec
+++ b/dist/redhat/scylla.spec
@@ -74,8 +74,8 @@ Requires:       %{product}-python3 = %{version}-%{release}
 AutoReqProv:    no
 Provides:       %{product}-tools:%{_bindir}/nodetool
 Provides:       %{product}-tools:%{_sysconfigdir}/bash_completion.d/nodetool-completion
-Provides:       scylla-enterprise = %{version}-%{release}
-Obsoletes:      scylla-enterprise < 2025.1.0
+Provides:       scylla-enterprise-server = %{version}-%{release}
+Obsoletes:      scylla-enterprise-server < 2025.1.0
 
 %description server
 This package contains ScyllaDB server.
@@ -220,6 +220,9 @@ Group:          Applications/Databases
 Summary:        Prometheus exporter for machine metrics
 License:        ASL 2.0
 URL:            https://github.com/prometheus/node_exporter
+Provides:       scylla-enterprise-node-exporter = %{version}-%{release}
+Obsoletes:      scylla-enterprise-node-exporter < 2025.1.0
+
 
 %description node-exporter
 Prometheus exporter for machine metrics, written in Go with pluggable metric collectors.
```

syuu1228 commented:
Also, I think `Obsoletes: $PKG_NAME < 2025.1.0` is probably not what we want, because RPM think 2025.1.0~rcX is LESS than 2025.1.0:
```
$ rpmdev-vercmp 2025.1.0~rc0-0.20250203.a20fa99ea38b 2025.1.0
2025.1.0~rc0-0.20250203.a20fa99ea38b < 2025.1.0
```
To avoid that, the version number need to be less than 2025.1.0~rcX, something like 2024.2.999 or  2024.3.0 or 2025.0.0.

yaronkaikov commented:
> Also, I think `Obsoletes: $PKG_NAME < 2025.1.0` is probably not what we want, because RPM think 2025.1.0~rcX is LESS than 2025.1.0:
> 
> ```
> $ rpmdev-vercmp 2025.1.0~rc0-0.20250203.a20fa99ea38b 2025.1.0
> 2025.1.0~rc0-0.20250203.a20fa99ea38b < 2025.1.0
> ```
> 
> To avoid that, the version number need to be less than 2025.1.0~rcX, something like 2024.2.999 or 2024.3.0 or 2025.0.0.

Why is this a problem?

yaronkaikov commented:
v4
- removed the submodule update commits. cqlsh already merged to master and python3 is waiting for https://github.com/scylladb/scylladb/pull/22517
- Added the manual test to the PR changelog

yaronkaikov commented:
@roydahan @fruch The rolling upgrade is doing the wrong process as it always removes the old release and installing from scratch, it should be fixed.  will raise an issue in SCT on Sunday

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/15091/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1151/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/3277/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/14411/)
:white_check_mark: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/135/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/6394/)
:white_check_mark: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/136/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/632/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/15091/)
#### Build Details:
- Duration: 4 hr 59 min
- Builder: spider5.cloudius-systems.com
